server/controllers/index.js

```diff
@@ -52,43 +54,65 @@ router.get('/auth/logout', auth, (req, res) => {
 });
 
 /* Create a room (a single instance of a game) */
-router.get('/api/room/create', auth, (req, res) => {
+router.post('/api/room/create', auth, (req, res) => {
     const { user } = req.session;
-    /* TODO: add check for when a room cannot be created and return an error */
     try {
-        let newGameState = new GameLogic(roomId);
-        let newGameRoom = {roomId, gameState: newGameState}
-        createRoom(user, roomId);
-        games.push(newGameRoom);
-        // Add the creator to the room
-        if (newGameState.canJoinGame(user)) { /* Game has not been started yet */
-            newGameState.joinGame(user, () => {
-                req.session.roomId = roomId;    
-                res.sendStatus(200);
-                roomId++;
-            });
-        } else { /* Game has already began */
-            res.status(400).json({message: "Game has already started."});
+        if (req.session.roomId !== undefined) {
+            const oldRoomId = req.session.roomId;
+            let game = getGameStateById(oldRoomId);
```

Comment:
Can this be a `const`?
```suggestion
            const game = getGameStateById(oldRoomId);
```